zk/stages/stage_sequence_execute.go

```diff
@@ -48,7 +48,7 @@ func SpawnSequencingStage(
 		return err
 	}
 
-	if lastSequence != nil && lastBatch < lastSequence.BatchNo && !cfg.zk.IsL1Recovery() {
+	if !cfg.zk.ShadowSequencer && lastSequence != nil && lastBatch < lastSequence.BatchNo && !cfg.zk.IsL1Recovery() {
```

Comment:
As a preference, could we add a new `if` statement to check this flag and panic if it's false so it doesn't get tied in with the base logic

Reply:
So like :
```
if lastSequence.... {
  if !cfg.zk.ShadowSequencer {
    panic()
```

Reply:
@hexoscott Jerry has just updated this line to make it even longer. Do you still find your change request valid or do you prefer it this way:
`	if lastSequence != nil && lastBatch < lastSequence.BatchNo && !(cfg.zk.IsL1Recovery() || cfg.zk.SequencerResequence || cfg.zk.ShadowSequencer) {`

Reply:
Ah yep I reviewed his PR earlier.  I'd say yes it's still valid because we're just checking if we're running in some special developer mode to stop it going into panic